File: tricore-server/src/routes/oa.rs
use actix_web::web;
use crate::handlers::oa;

pub fn configure(cfg: &mut web::ServiceConfig) {
    cfg.service(
        web::scope("/oa")
            // Employees
            .route("/employees", web::get().to(oa::list_employees))
            .route("/employees/{id}", web::get().to(oa::get_employee))
            .route("/employees", web::post().to(oa::create_employee))
            .route("/employees/{id}", web::put().to(oa::update_employee))
            .route("/employees/{id}", web::delete().to(oa::delete_employee))
            // Workflows
            .route("/workflows", web::get().to(oa::list_workflows))
            .route("/workflows/{id}", web::get().to(oa::get_workflow))
            .route("/workflows", web::post().to(oa::create_workflow))
            .route("/workflows/{id}", web::put().to(oa::update_workflow))
            .route("/workflows/{id}", web::delete().to(oa::delete_workflow))
            .route("/workflows/{id}/submit", web::post().to(oa::submit_workflow))
            .route("/workflows/{id}/steps", web::get().to(oa::get_steps))
            .route("/workflows/{id}/steps/{step_id}/review", web::post().to(oa::review_step))
            // Archives
            .route("/archives", web::get().to(oa::list_archives))
            .route("/archives/{id}", web::get().to(oa::get_archive))
            // Dashboard
            .route("/dashboard", web::get().to(oa::dashboard))
            // Templates
            .route("/templates", web::get().to(oa::list_templates))
            .route("/templates", web::post().to(oa::create_template))
            .route("/templates/{id}", web::put().to(oa::update_template))
            .route("/templates/{id}", web::delete().to(oa::delete_template)),
    );
}

File: tricore-web-rust/src/services/oa.ts
import api from './http'

export const oaAPI = {
  listEmployees: (params?: Record<string, unknown>) => api.get('/oa/employees', { params }),
  getEmployee: (id: string) => api.get(`/oa/employees/${id}`),
  createEmployee: (data: Record<string, unknown>) => api.post('/oa/employees', data),
  updateEmployee: (id: string, data: Record<string, unknown>) => api.put(`/oa/employees/${id}`, data),
  deleteEmployee: (id: string) => api.delete(`/oa/employees/${id}`),
  listWorkflows: (params?: Record<string, unknown>) => api.get('/oa/workflows', { params }),
  getWorkflow: (id: string) => api.get(`/oa/workflows/${id}`),
  createWorkflow: (data: Record<string, unknown>) => api.post('/oa/workflows', data),
  updateWorkflow: (id: string, data: Record<string, unknown>) => api.put(`/oa/workflows/${id}`, data),
  deleteWorkflow: (id: string) => api.delete(`/oa/workflows/${id}`),
  submitWorkflow: (id: string) => api.post(`/oa/workflows/${id}/submit`),
  getSteps: (id: string) => api.get(`/oa/workflows/${id}/steps`),
  reviewStep: (wfId: string, stepId: string, data: Record<string, unknown>) =>
    api.post(`/oa/workflows/${wfId}/steps/${stepId}/review`, data),
  listArchives: (params?: Record<string, unknown>) => api.get('/oa/archives', { params }),
  getArchive: (id: string) => api.get(`/oa/archives/${id}`),
  getDashboard: () => api.get('/oa/dashboard'),
  // Templates
  listTemplates: () => api.get('/oa/templates'),
  createTemplate: (data: Record<string, unknown>) => api.post('/oa/templates', data),
  updateTemplate: (id: string, data: Record<string, unknown>) => api.put(`/oa/templates/${id}`, data),
  deleteTemplate: (id: string) => api.delete(`/oa/templates/${id}`),
}
